### PR TITLE
Improve the-array-data-structure.md

### PR DESCRIPTION
**Pull request by kapnobatai137**
A user left this comment:
>"Or empty arrays of size..." - plese correct it - it sounds like you're trying to clear/delete the values in an array but actually it means to initialize an array without to place values in it. ;)

To remove any source of confusion, I have changed the phrase to say:
>Or, you can declare empty arrays of size 3 using:

Also, another user suggestion was to add color to code blocks. Before, all the array definitions were bunched in a single code block with no tag. I've separated each definition in its own code block, with the appropriate tags.

>Super great, one comment though would be to add coloured text to the code. For example, I don't know whether in JavaScript if "Array" is one of the "words" which has a specific meaning in JavaScript like "var" does. Great intro though :)

Here is how the code blocks display in the staging app:
test code block display

For some reason, the C++ code blocks are not getting any color. I've used the `c++` tag but I'm not sure that is the correct one.

### CODE REVIEW

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/comp-sci/data-structures-and-algorithms/intro-data-structures/the-array-data-structure.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/comp-sci/data-structures-and-algorithms/intro-data-structures/the-array-data-structure.md?ref=2832037f981930c9960efdabe7fb667c94828d46)

File: comp-sci/data-structures-and-algorithms/intro-data-structures/the-array-data-structure.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ parent: what-is-a-data-structure
 ---
 ## Content
 
-An **array** is a data structure organized as a collection of elements, for which **indexes** are used to access its content. The same information may appear in the array more than once, but under different indexes. This is how you could picture it:
+An **array** is a data structure organized as a collection of elements linearly, for which **indexes** are used to access its content. The same information may appear in the array more than once, but under different indexes. This is how you could picture it:
```
Comment (nem035) [inline comp-sci/data-structures-and-algorithms/intro-data-structures/the-array-data-structure.md]:
```suggestion
An **array** is a data structure organized as a linear collection of elements, for which **indexes** (numerical indicators for position) are used to access its content. The same information may appear in the array more than once, but under a different index. This is how you could picture it:
```

File: comp-sci/data-structures-and-algorithms/intro-data-structures/the-array-data-structure.md [hunk 2]
```diff
@@ -39,36 +39,49 @@ In this case, each book has its own *allocated space* and you can straight up *s
 A *dynamic* bookshelf would resize itself as soon as there is no more room for another book, while a *static* one wouldn't.
 
 
-You can declare an array using:
+You can construct an array using:
```
Comment (nem035) [inline comp-sci/data-structures-and-algorithms/intro-data-structures/the-array-data-structure.md]:
```suggestion
Here's how you can construct an array in a few languages:
```

File: comp-sci/data-structures-and-algorithms/intro-data-structures/the-array-data-structure.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ parent: what-is-a-data-structure
...
...-10.8l3%2010.8%203-10.8M431%2096v41m-3-10.8l3%2010.8%203-10.8%22%20stroke%3D%22currentColor%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22square%22%2F%3E%3Ctext%20font-family%3D%22RobotoMono-LightItalic%2C%20Roboto%20Mono%22%20font-size%3D%2230%22%20font-style%3D%22italic%22%20font-weight%3D%22300%22%20fill%3D%22currentColor%22%3E%3Ctspan%20x%3D%22115.15277%22%20y%3D%2280%22%3E0%3C%2Ftspan%3E%3C%2Ftext%3E%3Ctext%20font-family%3D%22RobotoMono-LightItalic%2C%20Roboto%20Mono%22%20font-size%3D%2230%22%20font-style%3D%22italic%22%20font-weight%3D%22300%22%20fill%3D%22currentColor%22%3E%3Ctspan%20x%3D%22219.15277%22%20y%3D%2280%22%3E1%3C%2Ftspan%3E%3C%2Ftext%3E%3Ctext%20font-family%3D%22RobotoMono-LightItalic%2C%20Roboto%20Mono%22%20font-size%3D%2230%22%20font-style%3D%22italic%22%20font-weight%3D%22300%22%20fill%3D%22currentColor%22%3E%3Ctspan%20x%3D%22321.15277%22%20y%3D%2280%22%3E2%3C%2Ftspan%3E%3C%2Ftext%3E%3Ctext%20font-family%3D%22RobotoMono-LightItalic%2C%20Roboto%20Mono%22%20font-size%3D%2230%22%20font-style%3D%22italic%22%20font-weight%3D%22300%22%20fill%3D%22currentColor%22%3E%3Ctspan%20x%3D%22423.15277%22%20y%3D%2280%22%3E3%3C%2Ftspan%3E%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fsvg%3E)
```
Comment (nem035) [inline comp-sci/data-structures-and-algorithms/intro-data-structures/the-array-data-structure.md]: Can we also use a PNG here?

Comment (kapnobatai137) [reply comp-sci/data-structures-and-algorithms/intro-data-structures/the-array-data-structure.md]: sure thing, I'll convert the images

**Comment by kapnobatai137** [comment]
I have also changed the color of the arrows and numbers (that aren't enclosed in boxes) to a dark-red. Before it was using `currentColor`, which was barely distinguishable on the dark theme.

**Pull request merged**